src/inmanta/references.py

```diff
@@ -219,21 +244,22 @@ def resolve_other[S: RefValue](self, value: "Reference[S] | S") -> S:
         When resolving reference on the compiler side, not all references will be resolved and using this method is required.
         """
         if isinstance(value, Reference):
-            return value.get()
+            return value.get(ctx)
         return value
 
     @classmethod
     def deserialize(
         cls: typing.Type[C],
         ref: SerializedReferenceLike,
         resource: "inmanta.resources.Resource",
+        ctx: "handler.LoggerABC",
```

Comment:
De-serialization is now a bit of a mixed bag: lower level reference are resolved during de-serialization, to level are not.

This is perhaps a bit weird. 

@bartv this is still part of what you built, I could factor it out in a two stage process (de-serialize with resolving, then resolve), but except from the slightly more complex behavior, there is not real disadvantage.

Reply:
I think this matters much. Mostly because we fail on the first non-found reference instead of trying to show them all.